system/lib/libc/sbrk.c

```diff
@@ -56,65 +58,48 @@ uintptr_t* emscripten_get_sbrk_ptr() {
 // Enforce preserving a minimal alignof(maxalign_t) alignment for sbrk.
 #define SBRK_ALIGNMENT (__alignof__(max_align_t))
 
+#ifdef __EMSCRIPTEN_SHARED_MEMORY__
+#define READ_SBRK_PTR(sbrk_ptr) (__c11_atomic_load((_Atomic(uintptr_t)*)(sbrk_ptr), __ATOMIC_SEQ_CST))
+#else
+#define READ_SBRK_PTR(sbrk_ptr) (*(sbrk_ptr))
```

Comment:
I don't think we need to make this distinction because `__c11_atomic_load` works fine, even on single-threads builds (it will just lower to a normal load).

We might even be able o avoid the `__c11_atomic_load` completely if we just give it the correct type since reads from `_Atomic` variables lower to `__c11_atomic_load`.

Reply:
Sure, updated

Reply:
> even on single-threads builds (it will just lower to a normal load).
Turns out that is not correct. It will cause an error
```
wasm-ld: error: --shared-memory is disallowed by sbrk.o because it was not compiled with 'atomics' or 'bulk-memory' features.
```

Reply:
Hmm.. we should be able to fix that.  I'll take a look after this lands.

Reply:
We can take care of this one we address #20803.  Resolve this here.